### PR TITLE
fix(TUP-19603)Suffix number is not proper when drag DB connection to a

### PR DESCRIPTION
**Pull request by jding-tlnd**
job
https://jira.talendforge.org/browse/TUP-19603

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/process/TAmazonOracleDndFilter.java [hunk 1]
```diff
@@ -47,18 +48,21 @@ public boolean except(Item item, ERepositoryObjectType type, RepositoryNode sele
                     }
                 }
                 Node node = new Node(component);
+                Process process = (Process) node.getProcess();
```
Comment (cmeng-talend) [inline main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/process/TAmazonOracleDndFilter.java]: better to check instance first:
IProcess proc = node.getProcess();
if (proc instanceof Process){
...

File: main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/process/TAmazonOracleDndFilter.java [hunk 2]
```diff
@@ -47,18 +48,21 @@ public boolean except(Item item, ERepositoryObjectType type, RepositoryNode sele
...
                         for (Object element : valuesList) {
                             String conType = EDatabaseTypeName.getTypeFromDbType(element.toString()).getDisplayName();
                             if (conType != null && dbType != null && conType.equals(dbType)) {
+                                process.removeNodeContainer(new NodeContainer(node));
                                 return false;
                             }
                         }
                     }
                 }
+                process.removeNodeContainer(new NodeContainer(node));
```
Comment (cmeng-talend) [inline main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/process/TAmazonOracleDndFilter.java]: better to use try finally block:
try {
    ...
} finally {
    process.removeNodeContainer(new NodeContainer(node));
}

File: main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/process/TAmazonOracleDndFilter.java (hunk 1 above)
Comment (jding-tlnd) [reply main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/process/TAmazonOracleDndFilter.java]: Solved, thanks

File: main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/process/TAmazonOracleDndFilter.java (hunk 2 above)
Comment (jding-tlnd) [reply main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/process/TAmazonOracleDndFilter.java]: Solved, thanks

**Pull request merged**